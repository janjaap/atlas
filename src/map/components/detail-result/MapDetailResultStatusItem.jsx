import React from 'react';
import PropTypes from 'prop-types';

const MapDetailResultStatusItem = ({ label, value, status }) => value && (
  <li className="map-detail-result__item">
    <section className="map-detail-result__item-content">
      <div className="map-detail-result__item-label">{label}</div>
      <div className={
          `map-detail-result__item-value
          ${status && status.length ?
            `map-detail-result__item-value--${status}` :
            ''
          }`
        }
      >
        {value}
      </div>
    </section>
  </li>
);

MapDetailResultStatusItem.defaultProps = {
  value: '',
  status: ''
};

MapDetailResultStatusItem.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  status: PropTypes.string
};

export default MapDetailResultStatusItem;
