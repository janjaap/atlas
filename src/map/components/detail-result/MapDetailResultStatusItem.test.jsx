import React from 'react';
import { shallow } from 'enzyme';

import MapDetailResultStatusItem
  from './MapDetailResultStatusItem';

describe('MapDetailResultItem', () => {
  it('should render label and value', () => {
    const label = 'label';
    const value = 'value';
    const status = '';
    const wrapper = shallow(
      <MapDetailResultStatusItem
        label={label}
        value={value}
        status={status}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render empty value', () => {
    const label = 'label';
    const value = '';
    const status = '';
    const wrapper = shallow(
      <MapDetailResultStatusItem
        label={label}
        value={value}
        status={status}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render without value', () => {
    const label = 'label';
    const wrapper = shallow(
      <MapDetailResultStatusItem
        label={label}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a red bullet if status is `alert`', () => {
    const label = 'label';
    const value = 'value';
    const status = 'alert';
    const wrapper = shallow(
      <MapDetailResultStatusItem
        label={label}
        value={value}
        status={status}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a blue bullet if status is `info`', () => {
    const label = 'label';
    const value = 'value';
    const status = 'info';
    const wrapper = shallow(
      <MapDetailResultStatusItem
        label={label}
        value={value}
        status={status}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
