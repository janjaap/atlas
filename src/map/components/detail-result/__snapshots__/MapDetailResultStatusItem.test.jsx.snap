// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailResultItem should not render empty value 1`] = `""`;

exports[`MapDetailResultItem should not render without value 1`] = `""`;

exports[`MapDetailResultItem should render a blue bullet if status is \`info\` 1`] = `
<li
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      label
    </div>
    <div
      className="map-detail-result__item-value
          map-detail-result__item-value--info"
    >
      value
    </div>
  </section>
</li>
`;

exports[`MapDetailResultItem should render a red bullet if status is \`alert\` 1`] = `
<li
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      label
    </div>
    <div
      className="map-detail-result__item-value
          map-detail-result__item-value--alert"
    >
      value
    </div>
  </section>
</li>
`;

exports[`MapDetailResultItem should render label and value 1`] = `
<li
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      label
    </div>
    <div
      className="map-detail-result__item-value
          "
    >
      value
    </div>
  </section>
</li>
`;
